projects/ngrx.io/tools/stackblitz-builder/builder.js

```diff
@@ -100,31 +102,68 @@ class StackblitzBuilder {
 
   _checkForOutdatedConfig() {
     // Ensure that nobody is trying to use the old config filenames (i.e. `plnkr.json`).
-    var plunkerPaths = path.join(this.basePath, '**/*plnkr.json');
-    var fileNames = globby.sync(plunkerPaths, { ignore: ['**/node_modules/**'] });
+    const plunkerPaths = path.join(this.basePath, '**/*plnkr.json');
+    const fileNames = globby.sync(plunkerPaths, {
+      ignore: ['**/node_modules/**'],
+    });
 
     if (fileNames.length) {
       const readmePath = path.join(__dirname, 'README.md');
       const errorMessage =
-          'One or more examples are still trying to use \'plnkr.json\' files for configuring ' +
-          'live examples. This is not supported any more. \'stackblitz.json\' should be used ' +
-          'instead.\n' +
-          `(Slight modifications may be required. See '${readmePath}' for more info.\n\n` +
-          fileNames.map(name => `- ${name}`).join('\n');
+        "One or more examples are still trying to use 'plnkr.json' files for configuring " +
+        "live examples. This is not supported any more. 'stackblitz.json' should be used " +
+        'instead.\n' +
+        `(Slight modifications may be required. See '${readmePath}' for more info.\n\n` +
+        fileNames.map((name) => `- ${name}`).join('\n');
 
       throw Error(errorMessage);
     }
   }
 
-  _createBaseStackblitzHtml(config) {
-    var file = '';
-
-    // TODO: Doesn't work properly yet
+  _getPrimaryFile(config) {
     if (config.file) {
-      file = `?file=${config.file}`;
+      if (!this._existsSync(path.join(config.basePath, config.file))) {
+        throw new Error(
+          `The specified primary file (${config.file}) does not exist in '${config.basePath}'.`
+        );
+      }
+      return config.file;
+    } else {
+      const defaultPrimaryFiles = [
+        'src/app/app.component.html',
+        'src/app/app.component.ts',
+        'src/app/main.ts',
+      ];
+      const primaryFile = defaultPrimaryFiles.find((fileName) =>
+        fs.existsSync(path.join(config.basePath, fileName))
+      );
+
+      if (!primaryFile) {
+        throw new Error(
+          `None of the default primary files (${defaultPrimaryFiles.join(
+            ', '
+          )}) exists in '${config.basePath}'.`
+        );
+      }
+
+      return primaryFile;
     }
-    var action = `https://run.stackblitz.com/api/angular/v1${file}`;
-    var html = `<!DOCTYPE html><html lang="en"><body>
+  }
+
+  _addStackblitzrc(postData) {
+    postData['project[files][.stackblitzrc]'] = JSON.stringify({
+      installDependencies: true,
+      startCommand: 'turbo start',
```

Comment:
It would be better to run tests by default for testing projects IMO.

We can add:

```json
{
  "projectType": "testing"
}
```

to `content/examples/testing-store/example-config.json`. Then set the start command (`turbo start` or `turbo test`) based on the `projectType`.

Reply:
yes good idea, I'll check this

Reply:
I added project type in `stackblitz.json` as we already load this file in builder. Let me know if you think that this is then wrong place